app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -41,6 +41,10 @@ def puppet?
   def check_override_selected
     return if (changed - ['description', 'override']).empty?
     return if override?
+    return if !override? && override_changed? && (changed - ['merge_overrides',
```

Comment:
> This is the current behavior on develop. I'd probably argue that this is a separate bug/issue.

Actually it does look like since we're setting merge_overrides, etc now when override changes, it bypasses line 42 so I'll fix that in this PR.